KIPs/kip-37.md

```diff
@@ -0,0 +1,746 @@
+---
+kip: 37
+title: Token Standard
+author: Kyungkoo Kai Kim <kai.kim@groundx.xyz>
+discussions-to: https://github.com/klaytn/kips/issues/37
+status: Draft
+type: Standards Track
+category: KCT
+created: 2020-08-19
+requires: 13
+---
+
+## Simple Summary
+
+A standard interface for contracts that manage multiple token types. A single deployed contract may include any combination of fungible tokens, non-fungible tokens or other configurations (e.g. semi-fungible tokens).
+
+## Abstract
+
+This standard outlines a smart contract interface that can represent any number of fungible and non-fungible token types. Existing standards such as KIP-7 require deployment of separate contracts per token type. The KIP-17 standard’s token ID is a single non-fungible index and the group of these non-fungibles is deployed as a single contract with settings for the entire collection. In contrast, the KIP-37 Multi Token Standard allows for each token ID to represent a new configurable token type, which may have its own metadata, supply and other attributes.
+
+The `_id` argument contained in each function’s argument set indicates a specific token or token type in a transaction.
+
+## Motivation
+
+Tokens standards like KIP-7 and KIP-17 require a separate contract to be deployed for each token type or collection. This places a lot of redundant bytecode on the Klaytn blockchain and limits certain functionality by the nature of separating each token contract into its own permissioned address. With the rise of blockchain games and platforms like Enjin Coin, game developers may be creating thousands of token types, and a new type of token standard is needed to support them. However, KIP-7 is not specific to games and many other applications can benefit from this flexibility.
+
+New functionality is possible with this design such as transferring multiple token types at once, saving on transaction costs. Trading (escrow / atomic swaps) of multiple tokens can be built on top of this standard and it removes the need to “approve” individual token contracts separately. It is also easy to describe and mix multiple fungible or non-fungible token types in a single contract.
+
+## Specification
+
+This document is heavily derived from [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155) written by Witek Radomski, Andrew Cooke, Philippe Castonguay, James Therien, Eric Binet, and Ronan Sandford.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Smart contracts implementing the KIP-37 standard MUST implement all of the functions in the KIP37 interface.
```

Comment:
Like other token standards, it would be very helpful to readers if this documents describes its differences from ERC-1155.

Here are references:
http://kips.klaytn.com/KIPs/kip-7#differences-with-erc-20
http://kips.klaytn.com/KIPs/kip-17#differences-from-erc-721

Reply:
So far there are no major differences with ERC-1155. However, I'm researching additional specification for KIP-37, after then I will add on more specification and differences on the document.